compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/ConstantValueResolver.java

```diff
@@ -468,4 +481,86 @@ private void checkUniqueness(BLangConstant constant) {
             }
         }
     }
+
+    private void updateSymbolType(BLangConstant constant) {
+        if (constant.symbol.kind == SymbolKind.CONSTANT && constant.symbol.type.getKind() != TypeKind.FINITE &&
+                constant.symbol.value != null) {
+            BTypeSymbol finiteTypeSymbol = Symbols.createTypeSymbol(SymTag.FINITE_TYPE,
+                    Flags.asMask(EnumSet.noneOf(Flag.class)), Names.EMPTY, constant.symbol.pkgID, null,
+                    constant.symbol.owner, constant.symbol.pos, VIRTUAL);
+            BFiniteType finiteType = new BFiniteType(finiteTypeSymbol);
+            BLangExpression expr;
+            expr = createExpression(constant.symbol.value.value, constant.symbol.type, constant.symbol.pos);
+            if (expr != null) {
+                finiteType.addValue(expr);
+                constant.symbol.type = finiteType;
+            }
+        }
+    }
+
+    private BLangExpression createExpression(Object value, BType type, Location pos) {
+        switch (type.getKind()) {
+            case INT:
+            case BYTE:
+            case FLOAT:
+            case DECIMAL:
+                return createConstantNumericLiteralExpression(value, type, pos);
```

Comment:
We can use `createConstantLiteralExpression` function. Instead of creating new function.

Reply:
If we do so, there type checking phase will log errors for numeric literals.